packages/palette_generator/lib/palette_generator.dart

```diff
@@ -157,34 +155,32 @@ class PaletteGenerator with Diagnosticable {
   /// The [imageProvider] and [timeout] arguments must not be null.
   static Future<PaletteGenerator> fromImageProvider(
     ImageProvider imageProvider, {
-    Size size,
-    Rect region,
-    int maximumColorCount,
-    List<PaletteFilter> filters,
-    List<PaletteTarget> targets,
+    Size? size,
+    Rect? region,
+    int? maximumColorCount,
+    List<PaletteFilter>? filters,
+    List<PaletteTarget>? targets,
```

Comment:
Same for most of these; do we want to allow explicitly passing null? Usually that makes APIs easier to use wrong, so default values are generally preferred.

Reply:
Done, except for region. Not sure what to put as default because there seems to be some logic involvement when it is `null` it is simply ignored but if it is `not null` then there is used some scaling. https://github.com/flutter/packages/blob/48a2148811a09b6354e8b02551e34d156ed56916/packages/palette_generator/lib/palette_generator.dart#L1126